mmseg/apis/inference.py

```diff
@@ -40,34 +44,50 @@ def init_model(config, checkpoint=None, device='cuda:0'):
     return model
 
 
-class LoadImage:
-    """A simple pipeline to load image."""
-
-    def __call__(self, results):
-        """Call function to load images into results.
+def _preprare_data(img, model: BaseSegmentor):
 
-        Args:
-            results (dict): A result dict contains the file name
-                of the image to be read.
+    cfg = model.cfg
+    if dict(type='LoadAnnotations') in cfg.test_pipeline:
+        cfg.test_pipeline.remove(dict(type='LoadAnnotations'))
+    # TODO: Consider using the singleton pattern to avoid building
+    # a pipeline for each inference
+    pipeline = Compose(cfg.test_pipeline)
+
+    if isinstance(img, str):
+        data, data_samples = model.data_preprocessor(
+            [pipeline(dict(img_path=img))], False)
```

Comment:
can we use `test_step` of model which can call `data_preprocessor` automatically 
https://github.com/open-mmlab/mmdetection/blob/a0a0211217edf66cd66b8e4074e45cc0f3dafbf7/mmdet/apis/inference.py#L137